### DISCUSSION TITLE
Getting standard tools to work with vsock mux UDS.

### DISCUSSION THREAD

**Discussion by ghost** [cloud-hypervisor/cloud-hypervisor#5182] [Q&A]
Is there a tool available (probably LD_PRELOAD hack) that would allow you to treat the vsock mux socket as a normal UDS?
Something that would automatically issue a CONNECT 1234\n and discard the OK before handing off the resultant standard UDS?

**Comment by rbradford** [comment]
Maybe `socat` could do that?

**Reply by ghost** [reply]
It's not even clear how to utilize socat to create a UDS which has already performed the vsock connection through the mux - and even if you succeed it will be a proxy imposing overhead.

One approach I can think of is a tool which would hijack connect() such that when connecting to the /path/v.sock mux it would automatically follow through with the non standard protocol.

**Reply by ghost** [reply]
I was able to find a solution which does just that and more besides: It's able to get software on the guest that uses INET/UNIX to use vsock instead. socat without the overhead.

https://github.com/kohlschutter/unsock

Might be helpful to others to reference this in documentation somewhere.
